entity/rule.php

```diff
@@ -519,28 +519,45 @@ public function set_anchor($anchor)
 	}
 
 	/**
-	* Get the language identifier
+	* Get the language iso
 	*
-	* @return int language identifier
+	* @return string language iso
 	* @access public
 	*/
 	public function get_language()
 	{
-		return isset($this->data['rule_language']) ? (int) $this->data['rule_language'] : 0;
+		return isset($this->data['rule_language']) ? $this->data['rule_language'] : 'en';
 	}
 
 	/**
-	 * Set the language identifier
+	 * Set the language iso
 	 *
-	 * @param int $language language identifier
+	 * @param string $language language iso
 	 * @return rule_interface $this object for chaining calls; load()->set()->save()
 	 * @access public
+	 * @throws \phpbb\boardrules\exception\unexpected_value
 	 */
 	public function set_language($language)
 	{
 		if (!isset($this->data['rule_language']))
 		{
-			$this->data['rule_language'] = (int) $language;
+			// Validate the requested language ISO is installed
+			if ($language !== '')
+			{
+				$sql = 'SELECT lang_id
+					FROM ' . LANG_TABLE . "
+					WHERE lang_iso = '" . $this->db->sql_escape($language) . "'";
+				$result = $this->db->sql_query($sql);
+				$lang_id = $this->db->sql_fetchfield('lang_id');
+				$this->db->sql_freeresult($result);
+
+				if (!$lang_id)
+				{
+					throw new \phpbb\boardrules\exception\unexpected_value(array('rule_language', 'WRONG_DATA_LANG', 'UNEXPECTED_VALUE'));
```

Comment:
```WRONG_DATA_LANG``` appears unhandled if you set wrong language iso, like
![rule_warn](https://cloud.githubusercontent.com/assets/212322/24071621/65b1f022-0c09-11e7-804b-ccf8c765f3cf.png)